scylla-macros/src/parser.rs

```diff
@@ -0,0 +1,22 @@
+use proc_macro::TokenStream;
+use proc_macro2::Ident;
+use syn::{parse2, Data, DeriveInput, Fields, FieldsNamed};
+
+/// Parsers the tokens_input to a DeriveInput and returns the struct name from which it derives and
+/// the named fields
+pub(crate) fn parse(tokens_input: TokenStream, current_derive: &str) -> (Ident, FieldsNamed) {
+    let input = parse2::<DeriveInput>(tokens_input.into()).expect("No DeriveInput");
```

Comment:
Is there a reason why `parse_macro_input!(tokens_input as DeriveInput);` was changed to `parse2::...`?
It looks like `parse_macro_input!` provides some error information on failure which I'm not sure if expect does:
https://github.com/dtolnay/syn/blob/fc37ee397164e2c0cc884b1f4949c7ebe5355246/src/parse_macro_input.rs#L109

Additionally `parse2` documentation states that it should be used with `TokenStream` from `proc_macro2` and when  using `TokenStream` from `proc_macro` `parse` should be used instead:
https://docs.rs/syn/1.0.57/syn/fn.parse2.html

Reply:
See: https://docs.rs/syn/1.0.57/syn/macro.parse_macro_input.html:

'This macro must be called from a function that returns proc_macro::TokenStream'

It triggers a compile error if that condition isn't met

Reply:
Oh okay that makes sense :+1:

We could also probably remove `.into()` since we are no longer converting from `proc_macro2::TokenStream` to `proc_macro::TokenStream`

Reply:
I added this change in a new PR: https://github.com/scylladb/scylla-rust-driver/pull/139. This branch had a lot of commits that I didn't want to have and I was unable to remove them... I'm not a git expert